tests/test_utils.py

```diff
@@ -220,6 +220,53 @@ def test_extract_image_from_dockerfile_from_with_build_args(self) -> None:
         # Should extract the variable reference
         self.assertEqual(result, '${BASE_IMAGE}')
 
+    def test_extract_image_from_dockerfile_with_resource_url(self) -> None:
+        """Test extracting Docker image using ResourceUrl scheme."""
+        dockerfile_content = """FROM alpine:3.18
+RUN apk add --no-cache python3
+"""
```

Comment:
NIT -- I find using textwrap.dedent more readable.

```suggestion
        dockerfile_content = textwrap.dedent("""
           FROM alpine:3.18
           RUN apk add --no-cache python3
        """)
```